### PR TITLE
Feature/api v2

### PR DESCRIPTION
**Pull request by jordaniza**
Deployment of an upgraded API experience for the Bountyboard. This release aims to give us production-grade CRUD functionality for a multitenant bounty architecture. It includes the following quality of life improvements:

**Strongly typed schema, validations and interfaces:**
1. Bounties and Customers strongly typed schema definitions with automatically generated typescript interfaces
2. Bounties and Customers serialisation and validation using Yup schema
3. Bounties, robust PATCH api protections, preventing accidental overwrites of nested objects
4. Bounties, polymorphic single schema object that automatically detects POST vs PATCH operations

**Bounties API**
1. API pagination with dynamic cursor
2. Addition of PATCH, POST, DELETE routes
3. Removal of the PUT route, in line with REST semantics (we can easily add this back if needed)

**Customers API**
1. Addition of PUT, DELETE, POST routes

**QA**
1. Backend test coverage > 80%, including integration test suite that connects to a local mongo instance

**Note**
While the functionality is complete, there is 1 pending issues I need to check:
1. Does the old API need supporting (content gateway), or will the current one still work for the team, the Bounties schema is unchanged for GET requests.

EDIT: Resolved the gating issue with a discord ID guard, this is intended to be lifted in subsequent developments with a proper auth module.

### CODE REVIEW

**Comment by jordaniza** [comment]
Added ultra basic gating to restrict to myself, behold and NS human

File: packages/react-app/next-env.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-// / <reference types="next" />
```
Comment (Behold3th) [inline packages/react-app/next-env.d.ts:1]: lets add to `.gitignore`

File: packages/react-app/src/pages/api/bounties/[id].ts [hunk 2]
```diff
@@ -1,90 +1,68 @@
...
-		},
-		body: JSON.stringify(embedMessage),
-	})
-		.then((response) => {
-			if (response.status !== 200) console.log(response);
-		})
-		.catch(console.error);
 };
+
+export default middlewares({ schema: BountySchema, handler });
```
Comment (Behold3th) [inline packages/react-app/src/pages/api/bounties/[id].ts:68]: I find this confusing. What about calling the validation and auth handers directly? I think it's chaining that breaks an expected pattern, I.e
```
await validate(handler)
await auth(handler)
await activity(handler)
```
vs validate(auth(activity(handler)))

File: packages/react-app/src/pages/api/customers/user.ts [hunk 3]
```diff
@@ -10,8 +10,8 @@ export default async function handler(
 	switch (req.method) {
 	case 'POST':
 		try {
-			const items = await service.getCustomersInUsersGuilds(req.body.guilds);
```
Comment (Behold3th) [inline packages/react-app/src/pages/api/customers/user.ts:13]: In general, why have this as an endpoint?

File: packages/react-app/src/services/bounty.service.ts [hunk 4]
```diff
@@ -57,19 +52,19 @@ export const getSortByValue = (originalInput: string): AcceptedSortOutputs => {
...
 };
 
-export const filterStatus = (query: FilterQuery, status?: string): FilterQuery => {
+export const filterStatus = (query: FilterQuery<BountyCollection>, status?: string): FilterQuery<BountyCollection> => {
 	/**
 	 * Pass status and append the corresponding status query to the query object
 	 */
 	if (status == null || status == '' || status == 'All' || status == undefined) {
-		query.status = ['Open', 'In-Progress', 'In-Review', 'Completed'];
+		query.status = { $in: ['Open', 'In-Progress', 'In-Review', 'Completed'] };
```
Comment (Behold3th) [inline packages/react-app/src/services/bounty.service.ts:60]: nice!

**Review by Behold3th — APPROVED** [review]
🔥 work, @jordaniza !

There's still some things I don't fully understand, and I've left some comments. However, no reason to block merging to develop.

**Comment by jordaniza** [comment]
Updated to base branch and merged

**Pull request merged**